pyanaconda/modules/subscription/runtime.py

```diff
@@ -360,3 +368,200 @@ def run(self):
             # is missing
             message = exception_dict.get("message", _("Failed to attach subscription."))
             raise SubscriptionError(message) from None
+
+
+class ParseAttachedSubscriptionsTask(Task):
+    """Parse data about subscriptions attached to the installation environment."""
+
+    def __init__(self, rhsm_entitlement_proxy, rhsm_syspurpose_proxy):
+        """Create a new attached subscriptions parsing task.
+
+        :param rhsm_entitlement_proxy: DBus proxy for the RHSM Entitlement object
+        :param rhsm_syspurpose_proxy: DBus proxy for the RHSM Syspurpose object
+        """
+        super().__init__()
+        self._rhsm_entitlement_proxy = rhsm_entitlement_proxy
+        self._rhsm_syspurpose_proxy = rhsm_syspurpose_proxy
+
+    @property
+    def name(self):
+        return "Parse attached subscription data"
+
+    @classmethod
+    def _pretty_date(self, date_from_json):
+        """Return pretty human readable date based on date from the input JSON."""
+        # fallback in case of the parsing fails
+        date_string = date_from_json
+        try:
+            # The start/end date in GetPools() output seems to be formatted as
+            # "Locale’s appropriate date representation.".
+            # See bug 1793501 for possible issues with RHSM provided date parsing.
+            date = datetime.datetime.strptime(date_from_json, "%m/%d/%y")
+            # get a nice human readable date
+            date_string = date.strftime("%b %d, %Y")
+        except ValueError:
+            log.warning("subscription: date parsing failed: %s", date_from_json)
+        return date_string
+
+    @classmethod
+    def _parse_subscription_json(cls, subscription_json):
```

Comment:
Switch this to `@staticmethod`.

Reply:
This needs to call `_pretty_date()`, so needs access to the class to call it in a clean way.